test/jdk/java/lang/Thread/virtual/ThreadPollOnYield.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test id=default
+ * @bug 8335269
+ * @summary Test that Thread.yield loop polls for safepoints
+ * @requires vm.continuations
+ * @library /test/lib
+ * @run junit/othervm ThreadPollOnYield
+ */
+
+/*
+ * @test id=c2
+ * @bug 8335269
+ * @summary Test that Thread.yield loop polls for safepoints
+ * @requires vm.continuations
+ * @library /test/lib
+ * @run junit/othervm -Xcomp -XX:-TieredCompilation -XX:CompileCommand=inline,*::yield* -XX:CompileCommand=inline,*::*Yield ThreadPollOnYield
```

Comment:
Given this forces -Xcomp shouldn't we skip running it when compilation mode is set via jtreg flags?

Reply:
The test should never fail even with external flags, so if anything it's just extra testing. But I can add vm.flagless if you prefer.

Reply:
flagless might be going too far as we won't test with other GC's etc. Can we just use `@requires vm.compMode != "Xcomp"` to exclude it from the Xcomp specific testing which is redundant.

Reply:
Done.